cogs/displayinfo.py

```diff
@@ -0,0 +1,116 @@
+except Exception as e:
+    print(e)
+
+# Select the database
+db = mongoClient.Rasenbot
+
+
+
+class DisplayInfo(commands.Cog):
+
+    def __init__(self,client):
+        self.client = client
+        self.timers = {}
+        self.current_collection = ""
+        self.current_guild_name = ""
+        self.count = 0
+        self.current_date = str(date.today())
+
+    # The 'mem_before' parameter represents the member's previous state, and 'mem_after' represents the current state.
+    @commands.Cog.listener()
+    async def on_member_update(self, mem_before:discord.Member, mem_after:discord.Member):
+        if self.current_guild_name != mem_before.guild.name:
+            if mem_before.guild.name not in db.list_collection_names():
+                newCollection = db[mem_before.guild.name]  # If the collection doesn't exist, create a new one for the guild/server
+                self.current_guild_name = mem_before.guild.name
+                self.current_collection = newCollection               
+                newCollection.insert_one({}) # Create a new blank document in the new Collection
+                self.timers = self.current_collection.find_one() # Load blank document from the MongoDB collection
+            else:
+                self.current_collection = db[mem_before.guild.name] # If the collection exists, get the existing collection for the guild
+                self.current_guild_name = mem_before.guild.name
+                self.timers = self.current_collection.find_one() # Load existing document from the MongoDB collection
+        
+        # Track activity updates for members
+        if not mem_before.activity and mem_after.activity: # activity starts/ongoing
+            if mem_after.name not in self.timers:
+                self.timers[mem_after.name] = {} # If the member is not already tracked, add them
+            if self.current_date not in self.timers[mem_after.name]:
+                self.timers[mem_after.name][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
+            if mem_after.activity.name not in self.timers[mem_after.name][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
+                self.timers[mem_after.name][self.current_date][mem_after.activity.name] = {
+                    "totalTime" : 0, 
+                    "startTime" : 0,
+                }
+            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["startTime"] = datetime.now() # Record the start time of the member's current activity
+            print("start:", self.timers)
+            asyncio.create_task(self.timed_update(mem_before, mem_after))
+        elif mem_before.activity and not mem_after.activity: # activity stopped
+            if mem_before.name in self.timers:
+                currentStartTime = self.timers[mem_before.name][self.current_date][mem_before.activity.name]["startTime"]
```

Comment:
Possibility of KeyError here? Might need a try/except in the case that the bot crashes/comes online when someone is already in an activity. When the `on_member_update` is triggered it could be that the activity is stopped but we don't have any tracking for this.